apps/ejabberd/src/ejabberd_c2s.erl

```diff
@@ -2412,6 +2414,38 @@ flush_messages(N, Acc) ->
               {N, Acc}
     end.
 
+
+%%%----------------------------------------------------------------------
+%%% XEP-0016
+%%%----------------------------------------------------------------------
+
+maybe_update_presence(StateData = #state{jid = JID}, NewList) ->
+    {Froms, _Tos, _Pending} = ejabberd_hooks:run_fold(
```

Comment:
This hook is not necessary here, I think. If I understand it correctly, all you need (subscription lists) are already in the `StateData` var. Take a look at fields `pres_*`, they are documented quite well in the [ejabberd_c2s.hrl](https://github.com/hippware/MongooseIM/blob/e5329f0d58963ab7875798ea01845aa6c016f99b/apps/ejabberd/src/ejabberd_c2s.hrl#L40-L55) file.